pkg/apis/template/extension.go

```diff
@@ -84,5 +84,22 @@ func (h Handler) RouteGetVersions(
 		return nil, 0, err
 	}
 
-	return model.ExposeTemplateVersions(entities), cnt, nil
+	// Set expose schema.
+	return exposeTemplateVersion(entities), cnt, nil
+}
+
+func exposeTemplateVersion(
+	entities []*model.TemplateVersion,
+) []*model.TemplateVersionOutput {
+	// Set expose schema.
+	for i, v := range entities {
+		if !v.CustomizeOpenAPISchema.IsEmpty() || v.Schema.IsEmpty() {
+			continue
+		}
+
+		entities[i].CustomizeOpenAPISchema = entities[i].Schema.
+			Expose()
+	}
+
+	return model.ExposeTemplateVersions(entities)
 }
```

Comment:
Can we merge CustomizedOpenAPISchema to entity.schema, and skip output of CustomizedOpenAPISchema in the API?

Reply:
Prefer using separated schemas and customizing the OpenAPI schema. 

At the API level, for list/get operations, you can simply ignore the details by using 'extract=-schema' and focus on viewing the customized OpenAPI schema. For the update operation, user should only set the 'customizeOpenAPISchema' and disregard the others. Both the list and put APIs use the same object, and the openapi schema is sufficient, eliminating the need for additional documentation.

At the database level, there is no need to modify the schema when updating the 'customizeOpenAPISchema'.